pkg/service/service.go

```diff
@@ -639,6 +658,33 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 	return new, lb.ID(svc.frontend.ID), nil
 }
 
+// filterServiceBackends returns the list of backends based on given front end ports.
+// The returned map will have key as port name/number, and value as list of respective backends.
+func filterServiceBackends(svc *svcInfo, onlyPorts []string) map[string][]lb.Backend {
+	if len(onlyPorts) == 0 {
+		return map[string][]lb.Backend{
+			anyPort: svc.backends,
+		}
+	}
+
+	res := map[string][]lb.Backend{}
+	for _, port := range onlyPorts {
+		// check for port number
+		if port == fmt.Sprintf("%d", svc.frontend.Port) {
```

Comment:
tiny optimization, this could be `fmt.Sprintf` could be replaced with a strings.Itoa or strings.Atoi